### PR TITLE
Feature/224 config by file

### LINKED ISSUE #135: Add unified exclude mechanisem
Currently there are specific exclude mechanisms for files and pom-plugins.
It would however be cool if one could disable every validation.

Idea:
* Each validation has a (generated) id. Examples: `required-files/README.md`, `pom-plugin:com.exasol:error-code-crawler`
* Each validation finding contains a hint on how to exclude it

### LINKED ISSUE #224: Move configuration to config file
Move configuration from the pom file to a dedicated project-keeper config file

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #224 
Closes #135

### CODE REVIEW

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/finding/FindingsUngrouperTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private static final SimpleValidationFinding FINDING_3 = SimpleValidationFinding.withMessage("my finding").build();
+
+    @Test
+    void testUngroup() {
+        final List<ValidationFinding> findings = List.of(FINDING_1,
+                new ValidationFindingGroup(List.of(FINDING_2, FINDING_3), () -> {
+                }));
+        final List<SimpleValidationFinding> ungroupFindings = new FindingsUngrouper().ungroupFindings(findings);
+        assertThat(ungroupFindings, Matchers.contains(FINDING_1, FINDING_2, FINDING_3));
+    }
```
Comment (kaklakariada) [inline project-keeper/src/test/java/com/exasol/projectkeeper/validators/finding/FindingsUngrouperTest.java:25]: Add a test with an empty findings list

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/finding/FindingsFixerTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        final SimpleValidationFinding finding1 = SimpleValidationFinding.withMessage("test 1").andFix(fix1).build();
+        final SimpleValidationFinding finding2 = SimpleValidationFinding.withMessage("test 2").andFix(fix2).build();
+        final SimpleValidationFinding finding3 = SimpleValidationFinding.withMessage("test 3").andFix(fix3).build();
+        final Logger logger = mock(Logger.class);
+        new FindingsFixer().fixFindings(
+                List.of(finding1, new ValidationFindingGroup(List.of(finding2, finding3), postFix)), logger);
+        verify(fix1).fixError(logger);
+        verify(fix2).fixError(logger);
+        verify(fix3).fixError(logger);
+        verify(postFix).run();
```
Comment (kaklakariada) [inline project-keeper/src/test/java/com/exasol/projectkeeper/validators/finding/FindingsFixerTest.java:29]: Use inOrder to verify that the fixes are executed in the correct order

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: project-keeper-maven-plugin/src/main/java/com/exasol/projectkeeper/plugin/AbstractProjectKeeperMojo.java [hunk 3]
```diff
@@ -33,12 +34,21 @@
...
     }
 
+    @Override
+    public final void execute() throws MojoFailureException {
+        if (isEnabled()) {
+            runProjectKeeper(createProjectKeeper());
+        }
+    }
+
+    protected abstract void runProjectKeeper(ProjectKeeper projectKeeper) throws MojoFailureException;
```
Comment (kaklakariada) [inline project-keeper-maven-plugin/src/main/java/com/exasol/projectkeeper/plugin/AbstractProjectKeeperMojo.java:56]: Even better 👍

**Review by kaklakariada — APPROVED** [review]
